### PR TITLE
Meta+Lagom: Enable CMAKE_BUILD_WITH_INSTALL_RPATH

### LINKED ISSUE #10055: Lagom: macOS Build does not work with -DBUILD_LAGOM=ON
Doing the standard Lagom build steps does not produce workable binaries in macOS

```
cmake -GNinja  -S Meta/Lagom -B Build/lagom -DBUILD_LAGOM=ON
ninja -C Build/lagom install
```

```
error: /Library/Developer/CommandLineTools/usr/bin/install_name_tool: for: /Users/mustafa/git/serenity/Build/superbuild-i686/../lagom-install/lib/liblagom-core.0.0.0.dylib (for architecture x86_64) option "-add_rpath @loader_path/../lib" would duplicate path, file already has LC_RPATH for: @loader_path/../lib
```

This is absoutely related to rpath handling, which is a super mystery to me (on macOS at least).

This resource: https://matthew-brett.github.io/docosx/mac_runtime_link.html
And the cmake reference: https://gitlab.kitware.com/cmake/community/-/wikis/doc/cmake/RPATH-handling

 don't make it immediately obvious how to get the `$ORIGIN:$ORIGIN/../${CMAKE_INSTALL_LIBDIR}"` behavior we want for parity with Linux given we're using GNUInstallDirs.

### PR DESCRIPTION
**Pull request by implicitfield**
Fixes #10055.

### CODE REVIEW

File: Meta/Lagom/CMakeLists.txt [hunk 1]
```diff
@@ -27,6 +27,12 @@ set(SerenityOS_SOURCE_DIR "${SERENITY_PROJECT_ROOT}" CACHE STRING "")
 
 set(SERENITY_CACHE_DIR "${PROJECT_BINARY_DIR}" CACHE STRING "")
 
+# Place binaries in a staging directory to ensure that rpaths always
+# remain valid.
+set(CMAKE_RUNTIME_OUTPUT_DIRECTORY "${SERENITY_PROJECT_ROOT}/Build/lagom/staging/bin")
+set(CMAKE_LIBRARY_OUTPUT_DIRECTORY "${SERENITY_PROJECT_ROOT}/Build/lagom/staging/${CMAKE_INSTALL_LIBDIR}")
+set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY "${SERENITY_PROJECT_ROOT}/Build/lagom/staging/${CMAKE_INSTALL_LIBDIR}")
```
Comment (ADKaster) [inline Meta/Lagom/CMakeLists.txt:34]: This breaks a lot of things. There's CI scripts, local scripts in Meta/, serenity.sh, etc that assume paths to Lagom binaries in the build directory.

Comment (ADKaster) [reply Meta/Lagom/CMakeLists.txt:34]: Hmm. well, this is the approach we use for jakt as well, and it seems to work there. so I think if we fixed all the places that have assumptions about build path to binaries, then this approach would be fine. Not sure about the `staging` directory though. That might be harder than it seems though.

**Comment by ADKaster** [comment]
Looking through this finally, There are two missed files:

https://github.com/SerenityOS/serenity/blob/30feb95d539c468259616c9bb6e260efcf571be5/.github/workflows/wasm.yml#L93-L94

and

https://github.com/SerenityOS/serenity/blob/30feb95d539c468259616c9bb6e260efcf571be5/.github/workflows/libjs-test262.yml#L128

I'm not sure exactly where the binary dir will be when serenity is FetchContent'd by libjs-test262/ , but surely it needs a bin/

An update to  libjs-262 will also be needed.

https://github.com/linusg/libjs-test262/blob/0bb4f5ede892afb41fa10603a7e7f91b5c9f412f/run_all_and_update_results.py#L32-L39

**Comment by awesomekling** [comment]
This breaks both macOS and Linux CI. From macOS runner:
```
Could not find executable /Users/runner/work/1/s/Meta/Lagom/Build/Ladybird/headless-browser
```

**Pull request merged**